### PR TITLE
Feature Selection and Regression Analyses

### PR DESCRIPTION
**Pull request by consta35**
Hi Amir,

I did feature selection and found 3 genes that look like they might influence the OFA1 (open field activity) Did I do this correctly?

Do I now put this into Random Forrests?

### CODE REVIEW

**Review by amirziai — CHANGES_REQUESTED** [review]
Great job aggregating the different methodologies. Now you have a subset of features that you can use to build models. For the next part you want to use a combination of model selection and hyper-parameter optimization in the grid or random search context (just like you did for the previous assignment).

I don't have the dataset so I couldn't inspect what is happening but somehow you've introduces columns `0` and `1` that didn't previously exist. You have to investigate what's going on.

**Comment by consta35** [comment]
rows 0 and 1 seem to appear after In [178]- where I write a for loop to get the mean of all the rows. I probably did something wrong there. Could you take a look? I have sent you the data set in the original email :)

Ok so now I want to model how the 3 genes predict behavior. This is non-binary so I don't want a classifier method. I used RandomForrestRegressor but don't know how to train it on subsets of my data and what not. What kind of regressor model should I use?

**Comment by consta35** [comment]
Ah Ok I fixed the 0 and 1 rows by doing this: result['Mean'] = result.mean(axis=1)

it works better. In the analysis I posted, I didn't scale the data. Should I? When I run OLS, the R2 is very low ~0.25. 

I think I should split the data into training groups and test groups and then run other regression models.

**Comment by amirziai** [comment]
Always split into train and test at the very least with the sklearn `train_test_split` [utility](http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html).

Once you have your train set do hyperparameter optimization as you did in the previous assignment. I recommend using [RandomizedSearchCV](http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.RandomizedSearchCV.html) in most cases. However in your case your data is pretty small so you can just use GridSearchCV and brute force all combinations of hyper-parameters.

Grid vs random search:
http://stats.stackexchange.com/questions/160479/practical-hyperparameter-optimization-random-vs-grid-search
https://medium.com/rants-on-machine-learning/smarter-parameter-sweeps-or-why-grid-search-is-plain-stupid-c17d97a0e881#.i87l7frsd

**Comment by amirziai** [comment]
Scaling is a good idea for linear regression, however for random forest it's not an issue.

**Comment by consta35** [comment]
Ok, I split my data into training and test sets and ran random forest regression. If I run it multiple times on the training set, I get a different score each time. Why is that? Also- I don't see a parameter for C in random forest regression. Hopefully we can talk about this in our meeting- but I'm making progress at least! :)
